packages/core/src/credentials/proofs/DataIntegrity.ts

```diff
@@ -158,6 +166,103 @@ export async function createProof<T>(
   return { ...document, proof }
 }
 
+const { byDid, byAlgorithm } = Signers.select
+
+/**
+ * Signs a document with a DID-related signer.
+ *
+ * @param args - Object holding all function arguments.
+ * @param args.document - An unsigned document. Any existing proofs will be overwritten.
```

Comment:
Yup, but for the current proof type this will update and change the document too, so it basically ends up being a different credential again.

Reply:
Ok, that is good to know. Maybe asking @ChrisChinchilla to have a read of this specifically here. To let people know that it would ultimately create a new credential.

Reply:
Updated the method description, maybe that helps?

Reply:
Oops, I did not properly check which file we are on here. This does in fact simply replace the proof while keeping the document intact - with the exception of context injection, which is a practice that is currently de-facto standard but kinda problematic. See https://github.com/digitalbazaar/data-integrity/issues/19 again.

Reply:
Interesting to read. So, currently no real solution until they deal with the issues you raised. It seems the conversation has gotten stale in the comments.